codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamFrame.java

```diff
@@ -50,11 +45,15 @@ public boolean equals(Object o) {
             return false;
         }
         Http2StreamFrame other = (Http2StreamFrame) o;
-        return streamId == other.streamId();
+        return stream == other.stream() || (stream != null && stream.equals(other.stream()));
     }
 
     @Override
     public int hashCode() {
```

Comment:
@Scottmitch if it can give "strange side-effects" I think I would prefer to just remove it. Are you strong about keeping it ?